tests/conftest.py

```diff
@@ -73,21 +78,46 @@ def playwright():
 
 
 @pytest.fixture(params=["chromium"])  # lets start with 1 browser for now, we can add more later
-def browser(
-    playwright: Playwright, request: SubRequest, run_server: Generator[str, Any, Any]
-) -> Generator[Page, Any, Any]:
+def browser(playwright: Playwright, request: SubRequest, get_server: str) -> Generator[Page, Any, Any]:
     browser = getattr(playwright, request.param).launch(headless=True)
-    context = browser.new_context(base_url=run_server)
+    context = browser.new_context(base_url=get_server)
     page = context.new_page()
 
     transport = httpx.HTTPTransport(retries=5, local_address="127.0.0.1")
     with httpx.Client(transport=transport, verify=False) as client:  # noqa: S501
-        client.get(f"{run_server}/", timeout=0.8)
+        client.get(f"{get_server}/", timeout=0.8)
 
     yield page
     browser.close()
 
 
+def pytest_addoption(parser: pytest.Parser):
+    parser.addoption("--server", action="store")
+    parser.addoption("--tad-version", action="store", default="0.1.0")
+
+
+@pytest.fixture(scope="session")
+def command_line_args(request: SubRequest) -> dict[str, Any]:
+    return {"server": request.config.getoption("--server"), "version": request.config.getoption("--tad-version")}
+
+
+@pytest.fixture(scope="session")
+def validate_version(command_line_args: dict[str, str], get_server: str) -> str:
+    transport = httpx.HTTPTransport(retries=5, local_address="127.0.0.1")
+    response_version = "Unknown"
+    for _x in range(10):
```

Comment:
why at a for loop? you can use the transport retry

Reply:
The server is (probably) already up running an 'older' version; I want to wait for the right version string to be returned.